sdk/tests/conformance2/rendering/draw-buffers.html

```diff
@@ -375,31 +375,37 @@
 
   checkAttachmentsForColor(attachments, [0, 255, 0, 255]);
 
-  // GLES3 spec section 3.9.2 Shader Outputs
-  debug("test that gl_FragColor only writes to color number zero");
+  // By default, gl_FragColor only writes to color number zero.
+  // drawBuffers can specify the color buffer that gl_FragColor should write into.
+  // See GLES3 spec section 4.2.1 Selecting Buffers for Writing.
+  debug("test that gl_FragColor writes to color buffer specified by drawBuffers");
   gl.bindFramebuffer(gl.FRAMEBUFFER, fb);
-  gl.drawBuffers(bufs);
+  var bufNum = 4;
+  var blueBufs = makeColorAttachmentArray(bufNum);
+  gl.drawBuffers(blueBufs);
   gl.useProgram(blueProgramESSL1);
```

Comment:
Excellent point Mo. This is a significant difference between ES 2.0 + EXT_draw_buffers, and ES 3.0 -- ESSL 1.00 shaders can't access more than one draw buffer in ES 3.0.

What is your reading of the specification in this area? Looking at page 186 of es_spec_3.0.4.pdf, it seems to me that if an ESSL 1.00 shader attempts to write to another index of gl_FragData other than zero, the behavior is not actually defined. The shader doesn't fail compilation, etc. This is briefly discussed in section 1.5 "Compatibility" of GLSL_ES_Specification_3.00.4.pdf. I don't know what the rules are about indexing gl_FragData by a variable rather than a constant. One way or the other, it seems that a clarification to the WebGL 2.0 spec is needed. What do you think that clarification should look like? Should it only allow indexing into gl_FragData by the constant 0 in ESSL 1.00 shaders in WebGL 2.0?